charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml

```diff
@@ -108,7 +108,7 @@ spec:
             - --thanos-ruler-instance-namespaces={{ .Values.prometheusOperator.thanosRulerInstanceNamespaces | join "," }}
             {{- end }}
             {{- if .Values.prometheusOperator.secretFieldSelector }}
-            - --secret-field-selector={{ .Values.prometheusOperator.secretFieldSelector }}
+            - --secret-field-selector={{ tpl (.Values.prometheusOperator.secretFieldSelector) . }}
```

Comment:
```suggestion
            - --secret-field-selector={{ tpl (.Values.prometheusOperator.secretFieldSelector) $ }}
```

Reply:
Can you tell me why '$' is recommended in this way? I found the '.' used in the helm documentation.
https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-tpl-function

Reply:
Sure, `.` is the current scope where `$` is the root scope so if you are in a range or a with, the current scope changes so .Values does not exist in there while `$` allows you to access the root scope

cf. https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with

Reply:
Also the other tpl are using the $ in this file so for homogeneity, let's use the same scheme